pkg/asset/installconfig/aws/permissions.go

```diff
@@ -325,6 +329,10 @@ var permissions = map[PermissionGroup][]string{
 		// Needed to filter zones by instance type
 		"ec2:DescribeInstanceTypeOfferings",
 	},
+	PermissionEdgeDefaultInstance: {
+		// Needed to filter zones by instance type
+		"ec2:DescribeInstanceTypeOfferings",
+	},
```

Comment:
@r4f4 @patrickdillon I would like to exercise a [little bit more this comment](https://github.com/openshift/installer/pull/9234#issuecomment-2498973132). Isn't `ec2:DescribeInstanceTypeOfferings` called every installer execution to discover types from the pool, and fallback to static ones **regardless** of the pool type? My impression that this permissions must be in the base set of required permissions, instead of only when edge zones is added.

Holding until hear from you all. Feel free to drop.
/hold

Reply:
> Not every execution, only when the instanceType is not set and/or when zones are not specified
Yeah, apologies. I mean default execution flow which those fields isn't set/required.

Reply:
>as it impacts managed services and their managed policies.

If this change is only to support managed services flow for now, LGTM.

Reply:
@patrickdillon yeah, I think a new bug is better for that case.

Reply:
Looking at this more carefully to write up the bug, it seems to me that [the behavior described here is not actually what happens:](https://github.com/openshift/installer/pull/9234#issuecomment-2498973132)
 
> Should the mostly users be penalty by getting more expensive, and slower instance types of mostly regions when some regions does not support it?

If I'm reading the code correctly, what happens is (generally), we always default to m6i and then we use the API to check if m6i is unavailable. If m6i is unavailable, we move on to checking for m5. If the API call fails, we will stick with m6i; so the risk is that without the permission, we would use m6i in a region where it is unavailable (not sure if those still exist?). 

I still think there is a valid case for making this perm required, but @mtulio @r4f4 could you confirm if I'm interpreting this correctly? The way I'm reading the code now is not what I originally thought and is not what is described above, so perhaps I'm misreading.

Reply:
@patrickdillon that's correct. The instance type list is hardcoded and the `TypeOfferings` call is just to pick the type from that list that's actually available in the specified region/zones. The function doc/comments even say so: https://github.com/openshift/installer/blob/master/pkg/asset/machines/worker.go#L211-L215

Reply:
Thanks, Patrick.

Reply:
@mtulio are you happy to proceede with this PR and make the perm always required as part of the new bug Patrick opened?

Reply:
> So IIUC the issue is that IFF you're installing to one of those regions & don't have this perm & don't specify an instance type, the install would fail (as it will default to the unavailable m6i instance).

Correct.

@r4f4 Yes. Patrick described the rationale behind my thoughts. When we introduced `m6i` we aimed better performance for the same price for control planes (in general new instances increases ~15%, and are cheaper, except 7th Gen). That time, m6i was available in a few regions, but the algorithm covered that edge cases.


> @mtulio are you happy to proceede with this PR and make the perm always required as part of the new bug Patrick opened?

Yes.
/lgtm

FWIW some time ago I wrote an [~~ugly~~ script][script] to collect and consolidate that information across zones around the globe, and mount the supported instance types on edge zones and in the region, if you are interested to explore, [here] is a start point, and sample [output][out].

[script]: https://github.com/mtulio/mtulio.labs/blob/devel/labs/ocp-aws-local-zones/aws-ec2-list-by-zone/list-instances-by-az.py
[out]: https://raw.githubusercontent.com/mtulio/mtulio.labs/refs/heads/devel/labs/ocp-aws-local-zones/aws-ec2-list-by-zone/output-aws-ec2-offering-table.txt